test/HcalDigiPipelineTest.cxx

```diff
@@ -0,0 +1,259 @@
+ * for your viewing?
+ */
+static const bool NTUPLIZE_ENERGIES = true;
+
+/**
+ * @class FakeSimHits
+ *
+ * Fills the event bus with an HcalSimHits collection with
+ * a range of energy hits. These hits are put into unique
+ * cells so that we can compare them to the correct energy
+ * in one event.
+ */
+class HcalFakeSimHits : public framework::Producer {
+
+  /**
+   * Maximum energy to make a simulated hit for [MeV]
+   */
+  const double maxEnergy_  = 3.*MIP_ENERGY; //100
+
+  /**
+   * Minimum energy to make a sim hit for [MeV]
+   * Needs to be above readout threshold (after internal HcalDigi's calculation)
+   */
+  //const double minEnergy_ = 1*MIP_ENERGY;
+  //const double minEnergy_ = 0.0685*MIP_ENERGY;
+  const double minEnergy_ = 0.24*MIP_ENERGY;
+  /**
+   * The step between energies is calculated depending on the min, max energy
+   * and the total number of sim hits you desire.
+   * [MeV]
+   */
+  const double energyStep_ = (maxEnergy_ - minEnergy_) / NUM_TEST_SIM_HITS;
+
+  /// current energy of the sim hit we are on
+  double currEnergy_ = minEnergy_;
+  
+public:
+  HcalFakeSimHits(const std::string &name, framework::Process &p)
+    : framework::Producer(name, p) {}
+  ~HcalFakeSimHits() {}
+
+  void beforeNewRun(ldmx::RunHeader &header) {
+    header.setDetectorName("ldmx-det-v12");
+  }
+
+  void produce(framework::Event &event) final override {
+    // put in a single sim hit
+    std::vector<ldmx::SimCalorimeterHit> pretendSimHits(1);
+
+    // back hcal, layer 8, strip 32 
+    ldmx::HcalID id(0,8,32);
+    pretendSimHits[0].setPosition( 84.7448, -208.116, 1223.11);
+    
+    // back hcal, layer 4, strip 30
+    //ldmx::HcalID id(0,4,30);
+    //pretendSimHits[0].setPosition( -9.90934, 14.4659, 1023.04); 
+
+    // back hcal, layer 5, strip 31
+    //ldmx::HcalID id(0,5,31);
+    //pretendSimHits[0].setPosition( -16.7397, 42.0896, 1079.69);
+
+    // back hcal, layer 9, strip 33
+    //ldmx::HcalID id(0,9,33);                                                                                                                                                                          
+    //pretendSimHits[0].setPosition( 0.160346,144.505, 1279.18);
+    
+    pretendSimHits[0].setID(id.raw());
+    pretendSimHits[0].addContrib(-1, //incidentID  
+				 -1, // trackID
+				 0, // pdg ID
+				 currEnergy_, // edep
+				 1. // time - 299mm is about 1ns from target and in middle of HCal
+    );
+
+    std::cout << "Testing Simhit: x " << pretendSimHits[0].getPosition()[0] << " y " << pretendSimHits[0].getPosition()[1] << " z " << pretendSimHits[0].getPosition()[2];
+    std::cout << " with id: section " << id.section() << " layer " << id.layer() << " strip " << id.strip();
+    std::cout << " and energy " << currEnergy_ << std::endl;
+    
+    // needs to be correct collection name
+    REQUIRE_NOTHROW(event.add("HcalSimHits", pretendSimHits));
+    std::cout << "here "<< std::endl;
```

Comment:
Same comment as the one below. Comment-out developer messages for future developers to un-comment.